set_choose.py

```
@@ -0,0 +1,28 @@
+import random
+from typing import TypeVar, Set
+T = TypeVar('T')
+
+
+#To note - the index reference is not constant
+#due to set
+def setChoice(s: Set[T]) -> T | None:
+    if len(s) == 0:
+        return None
```

nb. It may be desirable to raise an exception here rather than return none.